app/static/js/slcManagement.js

```diff
@@ -140,3 +144,101 @@ function unapproveProposal(el){
       }
     })
 }
+
+
+/************** Imported Courses Modal Stuff **************/
+
+function showAlterModalWithCourse(courseID) {
+  getImportedCourseInfo(courseID, function() {
+    $('#alterModal #alterCourseId').val(courseID);
+    var formAction = `/manageServiceLearning/imported/${courseID}`;
+
+    $('#alterModal form').on('submit', function(event) {
+      updateInstructorList(); // Fetch instructors from tr rows in Instructor Table before sending POST request
+      var dynamicRoute = `/manageServiceLearning/imported/${courseID}`;
+      $(this).attr('action', dynamicRoute);
+    });
+
+    $('#alterModal').modal('show');
+  });
+}
+
+
+function getImportedCourseInfo(courseID, callback) { // This function populates the fields in the modal of a chosen course with preexisting data
+  $.ajax({
+    url: `/manageServiceLearning/imported/${courseID}`,
+    type: "GET",
+    success: function(courseDict) {
+      if (Object.keys(courseDict).length !== 0){
+        // update the imported course modal
+        $('#instructorsTable').empty();
+        $('#courseName').val(courseDict['courseName']);
+        $('#courseAbbreviation').val(courseDict['courseAbbreviation']);
+        $('#hoursEarned').val(courseDict['hoursEarned']);
+        if (courseDict['instructors'] && courseDict['instructors'].length > 0) {
+          updateInstructorsTable(courseDict['instructors']);
+        }
+      }
+      if (callback) callback();
+    }
+  });
+}
+
+
+// Instructor manipulation functions
+// ---------------------------------
+
+function updateInstructorList() { // This function fetches instructor usernames and attached the list of usernames to the form submission
+  
+  $('#alterModal form input[name="instructor[]"]').remove(); // Clear existing hidden instructor inputs
+  var instructordata = getCourseInstructors(); // Get current instructors from the table
+
+  // Append new hidden inputs for each instructor
+  instructordata.forEach(function(instructor) {
+      $('<input>').attr({
+          type: 'hidden',
+          name: 'instructor[]',
+          value: instructor
+      }).appendTo('#alterModal form');
```

Comment:
The formatting here is pretty weird.

Reply:
It is to show that data are being added to a dictionary.